wtransport/src/tls.rs

```diff
@@ -57,18 +59,11 @@ impl Certificate {
     /// certificate by using the [`WebTransportOptions.serverCertificateHashes`] W3C API.
     ///
     /// [`WebTransportOptions.serverCertificateHashes`]: https://www.w3.org/TR/webtransport/#dom-webtransportoptions-servercertificatehashes
-    #[cfg(feature = "self-signed")]
-    #[cfg_attr(docsrs, doc(cfg(feature = "self-signed")))]
     pub fn hash(&self) -> Sha256Digest {
-        use ring::digest::digest;
-        use ring::digest::SHA256;
-
-        Sha256Digest(
-            digest(&SHA256, &self.0)
-                .as_ref()
-                .try_into()
-                .expect("SHA256 digest is 32 bytes len"),
-        )
+        // TODO(biagio): you might consider use crypto provider from new rustls version
+        let mut sha256 = Sha256::new();
+        sha256.update(self.der());
+        Sha256Digest(sha256.finalize().into())
```

Comment:
It's easier to just `Sha256Digest(Sha256::digest(self.der()))`

Reply:
Done thanks!